actix-web/src/introspection.rs

```diff
@@ -0,0 +1,263 @@
+use std::{
+    collections::HashMap,
+    fmt::Write as FmtWrite,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Mutex, OnceLock,
+    },
+    thread,
+};
+
+use serde::Serialize;
+
+use crate::http::Method;
+
+static REGISTRY: OnceLock<Mutex<IntrospectionNode>> = OnceLock::new();
+static DETAIL_REGISTRY: OnceLock<Mutex<HashMap<String, RouteDetail>>> = OnceLock::new();
+static DESIGNATED_THREAD: OnceLock<thread::ThreadId> = OnceLock::new();
+static IS_INITIALIZED: AtomicBool = AtomicBool::new(false);
```

Comment:
Using static global variables for shared state isn’t ideal, since all App instances in the same process will share the same data. I chose this approach for now to avoid bigger changes, but it should be improved later to support multiple independent App instances.